Use optimized_graph in graph_executor.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26703 Remove convert_to_ssa argument from runCleanupPasses - it is only used in one place.
* **#26705 Use optimized_graph in graph_executor.**

Differential Revision: [D17543281](https://our.internmc.facebook.com/intern/diff/D17543281)

Commit:
Use optimized_graph in graph_executor.

[ghstack-poisoned]

zdevito commented on torch/csrc/jit/function.h:
```diff
@@ -103,9 +103,13 @@ struct TORCH_API Function {
     if (executor_) {
       return executor_;
     }
+    // Make sure optimized graph is computed (do it before acquiring the lock to
```
This is a bit tricky, we should either switch the mutex to one that can be nested acquired by the same thread, or separate the public private API so that we can ask for the optimized graph below the lock. Otherwise, this will get confusing.

zdevito reviewed (approved):
Seems fine, but lets clean up the locking logic.

Commit:
Update on "Use optimized_graph in graph_executor."


Differential Revision: [D17543281](https://our.internmc.facebook.com/intern/diff/D17543281)

[ghstack-poisoned]
